ludwig/features/category_feature.py

```diff
@@ -198,13 +200,50 @@ def add_feature_data(
 class CategoryInputFeature(CategoryFeatureMixin, InputFeature):
     def __init__(self, input_feature_config: CategoryInputFeatureConfig, encoder_obj=None, **kwargs):
         super().__init__(input_feature_config, **kwargs)
+        if encoder_obj is None:
+            encoder_obj = self.initialize_encoder(input_feature_config.encoder)
+        self.encoder_obj = encoder_obj
 
-        if encoder_obj:
-            self.encoder_obj = encoder_obj
+        if input_feature_config.encoder.skip:
```

Comment:
We can abstract this stuff behind the `InputFeature` interface. It will be a big change, though, so I want to do it in a follow-up PR.